sig-robotics/proposal/Definition of cloud robot CRD and Operator/Definition of cloud robot CRD and Operator.md

```diff
@@ -0,0 +1,157 @@
+
+To achieve our goal, we need three kinds of CRDs: **robot CRD**, **task CRD** and **robotSync CRD**. The robot CRD and task CRD describe the robot status and task information respectively, and the robotSync CRD is used to record the number of robot CR and their configuration informations. When the key information of the robot changes, the status information must be reported immediately, otherwise the information is reported in the form of a heartbeat frame at a period set by the user. Task resources are created by users and delivered by the Controller.
+
+- The fields contained in the **robot** resource can be described as the table below:
+
+|        Fields       |   Type   |              Description               |
+| :-----------------: | :------: | :------------------------------------: |
+|       robotID       |   uint   |            Unique ID of a robot        |
+| registration_status | boolean  |  Indicates whether robot is registrated  |
+|      position       |  struct  | Indicates the location of the robot, including coordinates, the last point passed, etc. |
+|   running_status    |  struct  | Represents information related to robot motion, including linear velocity and angular velocity |
+|     task_info     |  struct  | Including the identification of orders and tasks, a task can be represented by a series of points and segments |
+|    battery_status    |  struct  | Including information such as remaining battery power, charging status, etc. |
+|   abnormal_events   | struct[] |         Record the exception information of the node         |
+|    robot_status     | boolean  |  indicates whether robot is ready for scheduling  |
+
+-  The fields contained in the task resource can be described as the table below:
+
+|          Field          |   Type   |                 Description                 |
+| :---------------------: | :------: | :-----------------------------------------: |
+|         orderID         |   uint   | The unique identification of the order in the system, used to identify whether multiple tasks belong to the same order |
+|         taskID          |   uint   | The orderID and taskID jointly identify the task status currently executed by the robot |
+|       allocated         |  string  | Empty if not assigned, otherwise the assigned robotID |
+|  point_status_sequence  | struct[] |     Sequence of points already received and not executed     |
+| segment_status_sequence | struct[] |    Sequence of segment already received and not executed     |
+| task_destination_ID  |  uint  | Describe the task destination point ID  |
+
+- The fields contained in the robotSync resource can be describe as follow:
+
+|          Field          |   Type   |             Description         |
+| :---------------------: | :------: | :-----------------------------: |
+|     registed_robots     | string[] | Keep a list of registered nodes, including nodes name and a list of node devices |
```

Comment:
The RobotID definition for the robot above is `uint`, why `string[]` here? Why not just use `node id` as robot id~

Reply:
I made a mistake here. It should be `struct[]` rather than `string[]`. Each struct contains a robotID as well as a list of sensors on the robot (that is, the device on this node). So I think this also resolves the first conversation. Sorry for my mistake.